packages/vue-client/src/components/GameView/PlayerSymbol.vue

```diff
@@ -0,0 +1,65 @@
+<script setup lang="ts">
+import { computed } from "vue";
+import TaegeukStone from "../TaegeukStone.vue";
+
+const props = defineProps<{
+  variant: string;
+  config: unknown;
+  gamestate: unknown;
+  player_nr: number;
+}>();
+
+const colors = computed(() => {
+  if (
+    [
+      "baduk",
+      "badukWithAbstractBoard",
+      "phantom",
+      "capture",
+      "tetris",
+      "pyramid",
+      "thue-morse",
+      "freeze",
+      "keima",
+      "drift",
+      "quantum",
+    ].includes(props.variant)
+  ) {
+    return [props.player_nr % 2 === 0 ? "black" : "white"];
+  }
+
+  if (props.variant === "chess") {
+    return [props.player_nr % 2 === 0 ? "white" : "black"];
+  }
+
+  if (props.variant === "fractional") {
+    // TODO: parse config to determine the players colors.
+    // unfortunately I can't figure out how to do it, because
+    // props.config is a reactive ref object. I don't understand why,
+    // because we also pass the config to the variantGameView.
```

Comment:
I am still wrapping my head around what is reactive and what's not in Vue, but I thought reactive objects can be treated mostly like non-reactive objects. The difference is that mutating them will trigger a re-render.

If you need to get a non-reactive version of the object, it looks like you can use `structuredClone` (though it seems a bit inefficient to create a proxy, then clone it back to non-proxy haha)

https://stackoverflow.com/questions/53662412/how-can-i-make-a-value-in-vue-js-non-reactive